index.js

```diff
@@ -99,15 +92,15 @@ function magnetURIDecode (uri) {
     result.urlList = result.urlList.concat(result.ws)
   }
 
+  result.peerAddresses = []
+  if (typeof result['x.pe'] === 'string' || Array.isArray(result['x.pe'])) {
+    result.peerAddresses = result.peerAddresses.concat(result['x.pe'])
+  }
+
   // remove duplicates by converting to Set and back
   result.announce = Array.from(new Set(result.announce))
   result.urlList = Array.from(new Set(result.urlList))
-
-  // convert x.pe into two level object
-  if (result['x.pe']) {
-    result.x = { pe: result['x.pe'] }
-    delete result['x.pe']
-  }
```

Comment:
I thought automatically turning x.pe into a nested object was clever but I don't think it's necessary. I renamed the nested x.pe object to the string 'x.pe' for a simpler implementation and less transformation back and forth.

The user is just going to use the friendly `peerAddresses` key anyway 👍